addons/l10n_tw_edi_ecpay/tests/test_edi.py

```diff
@@ -236,6 +236,47 @@ def test_07_fail_data_validation(self):
         with self.assertRaises(UserError):
             send_and_print.action_send_and_print()
 
+    def test_08_refund_invoice_with_payment(self):
```

Comment:
Thank you for the test, but from what I see, this wouldn't fail before the fix, would it?
You reproduce the flow, but never validate that the amounts are correct; and as you mock the return it doesn't test this change at all.